resource_customizations/monitoring.coreos.com/Prometheus/health.lua

```diff
@@ -1,23 +1,30 @@
-local hs={ status = "Progressing", message = "Waiting for initialization" }
+local hs = { status = "Progressing", message = "Waiting for initialization" }
+found_status = false 
 
 if obj.status ~= nil then
   if obj.status.conditions ~= nil then
     for i, condition in ipairs(obj.status.conditions) do
 
-      if condition.type == "Available" and condition.status ~= "True" then
-        if condition.reason == "SomePodsNotReady" then
-          hs.status = "Progressing"
+      if condition.type == "Available" then
+        found_status = true
+        if condition.status ~= "True" then
+            if condition.reason == "SomePodsNotReady" then
+              hs.status = "Progressing"
+            else
+              hs.status = "Degraded"
+            end
+            hs.message = condition.message or condition.reason
         else
-          hs.status = "Degraded"
+            hs.status = "Healthy"
+            hs.message = "All instances are available"
         end
-        hs.message = condition.message or condition.reason
-      end
-      if condition.type == "Available" and condition.status == "True" then
-        hs.status = "Healthy"
-        hs.message = "All instances are available"
       end
     end
   end
 end
 
-return hs
\ No newline at end of file
+if not found_status then
+    hs = { status = "Unknown", message = "Status is not provided" }
```

Comment:
In Argo CD, resources that don't provide status (like ConfigMap and Secrets), are always marked as healthy. The reason is that this impacts the overall Application status and in this case Apps will always be marked as `Unknown`. I suggest changing this to healthy instead.